<?xml version="1.0" ?>
<robot name="custom_robots" version="1.0">
  <material name="black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="grey_">
    <color rgba="0.2 0.2 0.2 1.0"/>
  </material>
  <material name="orange">
    <color rgba="1.0 0.423529411765 0.0392156862745 1.0"/>
  </material>
  <material name="brown">
    <color rgba="0.870588235294 0.811764705882 0.764705882353 1.0"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="metal">
    <color rgba="0.76862745098 0.749019607843 0.729411764706 1.0"/>
  </material>
  <material name="metal_grey">
    <color rgba="0.501960784314 0.501960784314 0.501960784314 1.0"/>
  </material>
  <material name="metal_white">
    <color rgba="0.866666666667 0.866666666667 0.890196078431 1.0"/>
  </material>
  <link name="base_link"/>
  <joint name="indy0_joint_base_link" type="fixed">
    <origin rpy="0.0 0.0 5.95316536284" xyz="1.84112781679 0.933416006597 0.64192495556"/>
    <parent link="base_link"/>
    <child link="indy0_link0"/>
  </joint>
  <link name="indy0_link0">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="1.59306955"/>
      <inertia ixx="0.00572623" ixy="2.51e-06" ixz="-0.0001138" iyy="0.00558959" iyz="-1.4e-07" izz="0.00966674"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://indy7_description/meshes/stl/Indy7_0.stl"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="-0.0 1.57079633 0.0" xyz="-0.1 0.0 0.03"/>
      <geometry>
        <cylinder length="0.2" radius="0.05"/>
      </geometry>
    </collision>
  </link>
  <link name="indy0_link1">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="11.8030102"/>
      <inertia ixx="0.15418559" ixy="-2.35e-06" ixz="1.739e-05" iyy="0.12937017" iyz="-0.04854267" izz="0.05964415"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://indy7_description/meshes/stl/Indy7_1.stl"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.072"/>
      <geometry>
        <cylinder length="0.3" radius="0.1"/>
      </geometry>
    </collision>
  </link>
  <link name="indy0_link2">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="7.99292141"/>
      <inertia ixx="0.2935698" ixy="-4e-07" ixz="1.441e-05" iyy="0.28094142" iyz="0.03727972" izz="0.03620609"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://indy7_description/meshes/stl/Indy7_2.stl"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="1.57079632679 0.0 -1.57079632679" xyz="-0.225 0.0 0.09"/>
      <geometry>
        <cylinder length="0.45" radius="0.073"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.01"/>
      <geometry>
        <cylinder length="0.2" radius="0.085"/>
      </geometry>
    </collision>
  </link>
  <link name="indy0_link3">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="2.99134127"/>
      <inertia ixx="0.03424593" ixy="1.49e-06" ixz="7.24e-06" iyy="0.03406024" iyz="0.00186009" izz="0.00450477"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://indy7_description/meshes/stl/Indy7_3.stl"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.03"/>
      <geometry>
        <cylinder length="0.2" radius="0.065"/>
      </geometry>
    </collision>
  </link>
  <link name="indy0_link4">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="2.12317035"/>
      <inertia ixx="0.00670405" ixy="3.75e-06" ixz="1.5e-06" iyy="0.00279246" iyz="-0.00127967" izz="0.00619341"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://indy7_description/meshes/stl/Indy7_4.stl"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.092"/>
      <geometry>
        <cylinder length="0.35" radius="0.06"/>
      </geometry>
    </collision>
  </link>
  <link name="indy0_link5">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="2.28865091"/>
      <inertia ixx="0.00994891" ixy="1.4e-07" ixz="3.21e-06" iyy="0.00978189" iyz="-0.00093546" izz="0.00271492"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://indy7_description/meshes/stl/Indy7_5.stl"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="-0.0 0.0 -0.0225"/>
      <geometry>
        <cylinder length="0.183" radius="0.055"/>
      </geometry>
    </collision>
  </link>
  <link name="indy0_link6">
    <inertial>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <mass value="0.40083918"/>
      <inertia ixx="0.00043534" ixy="1.3e-07" ixz="-2e-08" iyy="0.00044549" iyz="5.1e-07" izz="0.00059634"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://indy7_description/meshes/stl/Indy7_6.stl"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 -0.038"/>
      <geometry>
        <cylinder length="0.25" radius="0.06"/>
      </geometry>
    </collision>
  </link>
  <link name="indy0_tcp">
    <visual>
      <origin rpy="0.0 0.0 1.570796" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://my_mesh/meshes/stl/indy_gripper_asm2_res.STL"/>
      </geometry>
      <material name="metal_grey"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.005 0.044 0.1"/>
      <geometry>
        <cylinder length="0.08" radius="0.016"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="-0.005 0.044 0.1"/>
      <geometry>
        <cylinder length="0.08" radius="0.016"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="0.005 -0.044 0.1"/>
      <geometry>
        <cylinder length="0.08" radius="0.016"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0.0 0.0 0.0" xyz="-0.005 -0.044 0.1"/>
      <geometry>
        <cylinder length="0.08" radius="0.016"/>
      </geometry>
    </collision>
  </link>
  <joint name="indy0_joint0" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0775"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="indy0_link0"/>
    <child link="indy0_link1"/>
    <limit effort="3.14159265359" lower="-3.05432619099" upper="3.05432619099" velocity="1.308996939"/>
  </joint>
  <joint name="indy0_joint1" type="revolute">
    <origin rpy="1.57079632679 1.57079632679 0.0" xyz="0.0 -0.109 0.222"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="indy0_link1"/>
    <child link="indy0_link2"/>
    <limit effort="3.14159265359" lower="-3.05432619099" upper="3.05432619099" velocity="1.308996939"/>
  </joint>
  <joint name="indy0_joint2" type="revolute">
    <origin rpy="0.0 0.0 0.0" xyz="-0.45 0.0 -0.0305"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="indy0_link2"/>
    <child link="indy0_link3"/>
    <limit effort="3.14159265359" lower="-3.05432619099" upper="3.05432619099" velocity="1.308996939"/>
  </joint>
  <joint name="indy0_joint3" type="revolute">
    <origin rpy="-1.57079632679 0.0 1.57079632679" xyz="-0.267 0.0 -0.075"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="indy0_link3"/>
    <child link="indy0_link4"/>
    <limit effort="3.14159265359" lower="-3.05432619099" upper="3.05432619099" velocity="1.57079632679"/>
  </joint>
  <joint name="indy0_joint4" type="revolute">
    <origin rpy="1.57079632679 1.57079632679 0.0" xyz="0.0 -0.114 0.083"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="indy0_link4"/>
    <child link="indy0_link5"/>
    <limit effort="3.14159265359" lower="-3.05432619099" upper="3.05432619099" velocity="1.57079632679"/>
  </joint>
  <joint name="indy0_joint5" type="revolute">
    <origin rpy="-1.57079632679 0.0 1.57079632679" xyz="-0.168 0.0 0.069"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="indy0_link5"/>
    <child link="indy0_link6"/>
    <limit effort="3.14159265359" lower="-3.75245789179" upper="3.75245789179" velocity="1.57079632679"/>
  </joint>
  <joint name="indy0_tcp" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.06"/>
    <parent link="indy0_link6"/>
    <child link="indy0_tcp"/>
  </joint>
  <transmission name="indy0_tran0">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="indy0_joint0">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="indy0_motor0">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="indy0_tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="indy0_joint1">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="indy0_motor1">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="indy0_tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="indy0_joint2">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="indy0_motor2">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="indy0_tran3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="indy0_joint3">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="indy0_motor3">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="indy0_tran4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="indy0_joint4">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="indy0_motor4">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="indy0_tran5">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="indy0_joint5">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="indy0_motor5">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <joint name="panda1_joint_base_link" type="fixed">
    <origin rpy="0.0 -0.0 0.0" xyz="-0.0 -0.0 -0.333000004292"/>
    <parent link="panda1_link0"/>
    <child link="stem"/>
  </joint>
  <link name="panda1_link0">
    <visual>
      <origin rpy="0.0 -0.0 0.0" xyz="0.0 0.0 -0.333000004292"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link0.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="3.14159265 -1.37340076987 3.14159265" xyz="-0.0299999993294 0.0 -0.303000003099"/>
      <geometry>
        <cylinder length="0.102" radius="0.1"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="1.57079633 -0.0 0.0" xyz="-0.0500000007451 0.0 -0.303000003099"/>
      <geometry>
        <cylinder length="0.3" radius="0.05"/>
      </geometry>
    </collision>
  </link>
  <link name="panda1_link1">
    <visual>
      <origin rpy="1.57079632679 -0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link1.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="-1.14103405447 -0.0 0.0" xyz="0.0 0.0750000029802 -0.027499999851"/>
      <geometry>
        <cylinder length="0.132" radius="0.11"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda1_joint1" type="revolute">
    <origin rpy="1.57079632679 -0.0 0.0" xyz="-0.0 -0.0 -0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="panda1_link1"/>
    <child link="panda1_link0"/>
    <limit effort="3.14159265359" lower="-2.75" upper="2.75" velocity="1.308996939"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.75" soft_upper_limit="2.75"/>
  </joint>
  <link name="panda1_link2">
    <visual>
      <origin rpy="-1.57079632679 -0.0 0.0" xyz="0.0 1.54732216623e-12 -0.316000014544"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link2.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0.41241044501 -0.0 0.0" xyz="0.0 0.035000000149 -0.236000016332"/>
      <geometry>
        <cylinder length="0.175" radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda1_joint2" type="revolute">
    <origin rpy="-1.57079632679 -0.0 0.0" xyz="-0.0 1.54732216623e-12 -0.316000014544"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="panda1_link2"/>
    <child link="panda1_link1"/>
    <limit effort="3.14159265359" lower="-1.7" upper="1.7" velocity="1.308996939"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7" soft_upper_limit="1.7"/>
  </joint>
  <link name="panda1_link3">
    <visual>
      <origin rpy="-1.57079632679 -0.0 0.0" xyz="-0.0825000032783 0.0 0.0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link3.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="-0.837979039329 -0.827091928922 1.70460287535" xyz="-0.0425000041723 -0.035000000149 -0.0270000007004"/>
      <geometry>
        <cylinder length="0.12" radius="0.09"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda1_joint3" type="revolute">
    <origin rpy="-1.57079632679 -0.0 0.0" xyz="-0.0825000032783 -0.0 -0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="panda1_link3"/>
    <child link="panda1_link2"/>
    <limit effort="3.14159265359" lower="-2.75" upper="2.75" velocity="1.308996939"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.75" soft_upper_limit="2.75"/>
  </joint>
  <link name="panda1_link4">
    <visual>
      <origin rpy="1.57079632679 -0.0 0.0" xyz="0.0825000032783 -1.88029019438e-12 -0.384000003338"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link4.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0.426982301215 -0.719163938304 -1.0572056574" xyz="0.027000002563 -0.0270000007004 -0.326000005007"/>
      <geometry>
        <cylinder length="0.17" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda1_joint4" type="revolute">
    <origin rpy="1.57079632679 -0.0 0.0" xyz="0.0825000032783 -1.88029019438e-12 -0.384000003338"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="panda1_link4"/>
    <child link="panda1_link3"/>
    <limit effort="3.14159265359" lower="0.1" upper="2.9" velocity="1.308996939"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="0.1" soft_upper_limit="2.9"/>
  </joint>
  <link name="panda1_link5">
    <visual>
      <origin rpy="-1.57079632679 -0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link5.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.30344094827 -0.0 0.0" xyz="0.0 -0.115000002086 -0.0315000005066"/>
      <geometry>
        <cylinder length="0.238" radius="0.08"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda1_joint5" type="revolute">
    <origin rpy="-1.57079632679 -0.0 0.0" xyz="-0.0 -0.0 -0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="panda1_link5"/>
    <child link="panda1_link4"/>
    <limit effort="3.14159265359" lower="-2.75" upper="2.75" velocity="1.57079632679"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.75" soft_upper_limit="2.75"/>
  </joint>
  <link name="panda1_link6">
    <visual>
      <origin rpy="-1.57079632679 -0.0 0.0" xyz="-0.0879999995232 0.0 0.0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link6.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="-1.57079632679 -4.89658877681e-12 1.57079633" xyz="-0.0529999993742 0.0 0.0"/>
      <geometry>
        <cylinder length="0.09" radius="0.065"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="3.21000004355e-09 -0.0 0.0" xyz="0.00200000405312 -4.89658870905e-14 0.00999999977648"/>
      <geometry>
        <cylinder length="0.13" radius="0.05"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="-1.37079633217 4.89658877681e-12 -1.57079633" xyz="-0.00299999862909 2.4482943884e-13 -0.0500000007451"/>
      <geometry>
        <cylinder length="0.08" radius="0.04"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda1_joint6" type="revolute">
    <origin rpy="-1.57079632679 -0.0 0.0" xyz="-0.0879999995232 -0.0 -0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="panda1_link6"/>
    <child link="panda1_link5"/>
    <limit effort="3.14159265359" lower="-3.6" upper="-0.1" velocity="1.57079632679"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.6" soft_upper_limit="-0.1"/>
  </joint>
  <link name="panda1_link7">
    <visual>
      <origin rpy="0.0 -0.0 0.0" xyz="0.0 0.0 -0.107000000775"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/link7.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="panda1_joint7" type="revolute">
    <origin rpy="0.0 -0.0 0.0" xyz="-0.0 -0.0 -0.107000000775"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="panda1_link7"/>
    <child link="panda1_link6"/>
    <limit effort="3.14159265359" lower="-2.75" upper="2.75" velocity="1.57079632679"/>
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.75" soft_upper_limit="2.75"/>
  </joint>
  <link name="panda1_link8"/>
  <joint name="panda1_joint8" type="fixed">
    <origin rpy="0.0 -0.0 0.785398163397" xyz="-0.0 -0.0 -0.0"/>
    <axis xyz="0.0 0.0 0.0"/>
    <parent link="panda1_link8"/>
    <child link="panda1_link7"/>
  </joint>
  <joint name="panda1_hand_joint" type="fixed">
    <origin rpy="0.0 -0.0 0.0" xyz="0.0 0.0 0.0"/>
    <parent link="panda1_hand"/>
    <child link="panda1_link8"/>
  </joint>
  <link name="panda1_hand">
    <visual>
      <origin rpy="0.0 -0.0 0.0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/hand.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="1.57079633 -0.0 0.0" xyz="0.0 0.0 0.0299999993294"/>
      <geometry>
        <cylinder length="0.16" radius="0.045"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="-0.0 1.57079632359 0.0" xyz="0.0 0.070000000298 -0.019999999553"/>
      <geometry>
        <cylinder length="0.055" radius="0.02"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="1.57079633 -0.0 0.0" xyz="0.0 -0.070000000298 -0.019999999553"/>
      <geometry>
        <cylinder length="0.05" radius="0.01"/>
      </geometry>
    </collision>
  </link>
  <link name="panda1_leftfinger">
    <visual>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="-2.73268482 0.0 0.0" xyz="0.0 0.0135 0.035"/>
      <geometry>
        <cylinder length="0.0325" radius="0.018"/>
      </geometry>
    </collision>
  </link>
  <link name="panda1_rightfinger">
    <visual>
      <origin rpy="0.0 0.0 3.14159265359" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://franka_description/meshes/visual/finger.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="-2.73268482 0.0 -3.14159265" xyz="0.0 -0.0135 0.035"/>
      <geometry>
        <cylinder length="0.0325" radius="0.018"/>
      </geometry>
    </collision>
  </link>
  <joint name="panda1_finger_joint1" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.04 0.0584"/>
    <parent link="panda1_hand"/>
    <child link="panda1_leftfinger"/>
  </joint>
  <joint name="panda1_finger_joint2" type="fixed">
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.04 0.0584"/>
    <parent link="panda1_hand"/>
    <child link="panda1_rightfinger"/>
    <mimic joint="panda1_finger_joint1"/>
  </joint>
  <link name="stem"/>
  <joint name="stem_joint_base_link" type="fixed">
    <origin rpy="0.0 -0.0 -1.0299310913" xyz="-1.68474316597 0.03768658638 -0.960543334484"/>
    <parent link="stem"/>
    <child link="base_link"/>
  </joint>
</robot>